### ISSUE TITLE
Azure OpenAI Errors are unhandled

### ISSUE THREAD

**Issue by xangelix** [64bit/async-openai#94]
Here's one example that only gives a JSONDeserialize error.

```json
{
  "error": {
    "code": 429,
    "message":  "Requests to the Creates a completion for the chat message Operation under Azure OpenAI API version 2023-05-15 have exceeded token rate limit of your current OpenAI S0 pricing tier. Please retry after 60 seconds. Please go here: https://aka.ms/oai/quotaincrease if you would like to further increase the default rate limit."
  }
}
```

JSONDeserialize(Error("missing field `type`"))

**Comment by 64bit** [comment]
Thank you @xangelix for reporting the issue. 

This is because of unfortunate differences in specs for Error:
-   https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-03-15-preview/inference.json#L774C25-L796
- https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L1579-L1598

Quick fix could be making `r#type` field in `ApiError` as `Option<String>`.

**Comment by ChristopherJMiller** [comment]
This is affecting me as well, I'll work on at least implementing the quick fix, and come back to it if more should be done 😃

Ideally from an end-library-user perspective I want the error visible so I know what's wrong

**Comment by 64bit** [comment]
Resolved by #95
